stdlib/transactions/src/main/ballerina/transactions/transaction_block.bal

```diff
@@ -14,6 +14,182 @@
+    boolean isTrxSuccess = false;
+    int rCnt = 1;
+    if (rMax < 0) {
+        error err = error("TransactionError", message = "invalid retry count");
+        panic err;
+    }
+    // If global tx enabled, it is managed via transaction coordinator.Otherwise it is managed locally
+    // without any interaction with the transaction coordinator.
+    if (isNestedTransaction()) {
+        // Starting a transaction within already infected transaction.
+        error err = error("{ballerina}TransactionError", message =  "dynamically nested transactions are not allowed");
+        panic err;
+    }
+    TransactionContext|error txnContext;
+    string transactionId = "";
+    int|error trxResult = -1;
+    error? abortResult = ();
+    error? committedResult = ();
+    error? retryResult = ();
+    error
+    ? rollbackResult = ();
+    
+    while (true) {
+        txnContext =  beginTransaction((), transactionBlockId, "", TWO_PHASE_COMMIT);
+        if (txnContext is error) {
+            panic txnContext;
+        } else {
+            transactionId = txnContext.transactionId;
+            setTransactionContext(txnContext);
+        }
+        trxResult = trap trxFunc.call();
+        if (trxResult is int) {
+            // If transaction result == 0, means it is successful.
+            if (trxResult == 0) { 
+                var endSuccess = trap endTransaction(transactionId, transactionBlockId);
+                if (endSuccess is string) {
+                    if (endSuccess == OUTCOME_COMMITTED) {
+                        isTrxSuccess = true;
+                        break;
+                    }
+                }
+            }
+            // If transaction result == -1, means it was aborted.
+            if (trxResult == -1) { // abort
+                abortResult = trap abortTransaction(transactionId, transactionBlockId);
+                break;
+            }
+        } else {
+            log:printDebug(trxResult.reason());
+        }
+        rCnt = rCnt + 1;
+        // retry
+        if (rCnt <= rMax) {
+            rollbackResult = trap rollbackTransaction(transactionBlockId);
+            if (rollbackResult is error) {
+                log:printDebug(rollbackResult.reason());
+            }
+            retryResult = trap retryFunc.call();
+            if (retryResult is error) {
+                log:printDebug(retryResult.reason());
+            }
+        } else {
+            break;
+        }
+    }
+    if (isTrxSuccess) {
+        committedResult = trap committedFunc.call();
+        if (committedResult is error) {
+            log:printDebug(committedResult.reason());
+        }
+    } else {
+        abortResult = trap handleAbortTransaction(transactionId, transactionBlockId, abortedFunc);
+        if (abortResult is error) {
+            log:printDebug(abortResult.reason());
+        }
+    }
+    cleanupTransactionContext(transactionBlockId);
+    if (trxResult is error) {
```

Comment:
This is re-throwing panic error captured in transaction block right? Shall we add a comment documenting that.